base/string.jl

```diff
@@ -1235,6 +1235,49 @@ print_shell_escaped(io::IO) = nothing
 
 shell_escape(args::AbstractString...) = sprint(print_shell_escaped, args...)
 
+# Functions for Windows CMD-shell escaping
+function print_cmdshell_word(io::IO, word::AbstractString)
+    if isempty(word)
+        print(io, "\"\"")
+    end
+    if length(word)==1
+        if word[1]=='|' || word[1]=='<' || word[1]=='>'
+            print(io, word)  # don't escape single pipe character
+            return
+        end
+    end
+    special_dq    = " \t,;="  # delimiters can't be quoted with "^"
+    special_caret = "\"\n^&|<>()"
```

Comment:
i think it might be better to throw an error if the passed command line includes a \n, since afaict you can't create a command line that includes an embedded `\n` (or `\r`) symbol:

```
If it is a caret (^) the next character has no special meaning, the caret itself is removed, if the caret is the last character of the line, the next line is appended, the first charater of the next line is always handled as escaped charater.
```